### PR TITLE
[Feature] Add accurate fp32 SFPU path for ttnn.min and ttnn.sum via fast_and_approximate_mode

### LINKED ISSUE #32274: Generic reductions(min, max, mean, ...) over all dimensions return wrong result
### Describe the bug

Generic reduction operations(min, max, mean, ...) over all dimensions produce the wrong result compared to the torch output.

### Steps to reproduce the issue

This Python test should reproduce the issue:
```Python
# ttnn.max example
import torch
import ttnn

device = ttnn.open_device(device_id=0)

torch_input = torch.randn((32, 32), dtype=torch.float32)
ttnn_input = ttnn.from_torch(torch_input, device=device, layout=ttnn.TILE_LAYOUT)

ttnn_output = ttnn.max(ttnn_input, None, False, memory_config=ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None))
torch_output = torch.max(torch_input)

print("Input tensor:", torch_input)
print("TTNN Output:", ttnn_output)
print("Torch Output:", torch_output)

all_close = torch.allclose(ttnn.to_torch(ttnn_output), torch_output, rtol=1e-05, atol=1e-08)
print("Outputs are close:", all_close)

ttnn.close_device(device)
```

```Python
#ttnn.min example
import torch
import ttnn

device = ttnn.open_device(device_id=0)

torch_input = torch.randn((32, 32), dtype=torch.float32)
ttnn_input = ttnn.from_torch(torch_input, device=device, layout=ttnn.TILE_LAYOUT)

ttnn_output = ttnn.min(ttnn_input, None, False, memory_config=ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None))
torch_output = torch.min(torch_input)

print("Input tensor:", torch_input)
print("TTNN Output:", ttnn_output)
print("Torch Output:", torch_output)

all_close = torch.allclose(ttnn.to_torch(ttnn_output), torch_output, rtol=1e-05, atol=1e-08)
print("Outputs are close:", all_close)

ttnn.close_device(device)
```

### Expected behavior

The above allclose() checks should pass.

### Please complete the following environment information

- Ubuntu 22.04
- Latest main

### PR DESCRIPTION
**Pull request by ign-sudha**
### PR Category
- Feature

### Summary
- Fixes #32274.

FLOAT32 generic reductions on the FPU truncate operands to TF32. For `min`/`max` that can return a value not in the input; for `sum`/`mean` the accumulator is lossy.

`fast_and_approximate_mode` selects the path:

| Mode | Default | Path | Precision |
| --- | --- | --- | --- |
| `False` | yes | SFPU | full fp32 |
| `True` | | FPU | TF32 (faster) |

Falls back to FPU when `fp32_dest_acc_en=False` or on Quasar. No effect for non-FLOAT32.

Accurate SFPU coverage (`fast_and_approximate_mode=False`):

| | `ttnn.max` | `ttnn.mean` | `ttnn.min` | `ttnn.sum` |
| --- | --- | --- | --- | --- |
| Single-axis (W / H / HW) | #52718 | #48896 | this PR | this PR |
| Multi-axis / `dim=None` | #52718 | this PR | this PR | this PR |

- #48896 put FLOAT32 mean on SFPU as SUM + 1/N, but only for a single W/H/HW dispatch. Multi-axis / `dim=None` (rank > 2) still ran per-axis sub-steps on the FPU. This PR does forward the caller’s mode through those mean sub-steps.

Default FLOAT32 `min`/`sum` (and multi-axis `mean`) are slower than before. Pass `fast_and_approximate_mode=True` to keep the old FPU path.

### Notes for reviewers

- **MIN** - Accurate fp32 min drives the LLK MIN reduce directly, like Int32 min. Fast-mode float/bf16 min still lowers to `-MAX(-x)`. Both negative-scalar flips (`max→min`, `min→max`) forward the flag.

- **SUM / mean** - Mean already lowered to SUM + post-mul `1/N` (#48896). Dense RM mean/sum keep the same SFPU fold.

- **Multi-axis** - Every decomposition here is exact (sum of sums, max of maxes, min of mins), so sub-steps carry the caller's mode.

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:ign/sudha-generic-reductions-fastandapproxmode)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:ign/sudha-generic-reductions-fastandapproxmode)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ign/sudha-generic-reductions-fastandapproxmode)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:ign/sudha-generic-reductions-fastandapproxmode)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:ign/sudha-generic-reductions-fastandapproxmode)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:ign/sudha-generic-reductions-fastandapproxmode)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:ign/sudha-generic-reductions-fastandapproxmode)

### CODE REVIEW

**Review by s-jovic — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/reduction/generic/device/reduce_op_multi_core_h_program_factory.cpp
Comment (iwroszTT) [inline ttnn/cpp/ttnn/operations/reduction/generic/device/reduce_op_multi_core_h_program_factory.cpp:383]: On the RM path cb_acc (c_5) is used to accumulated partials. Should we also have:
`unpack_to_dest_mode[CBIndex::c_5] = UnpackToDestMode::UnpackToDestFp32;`

This should allow to tighten thresholds in the test_row_major_reduce.py:
`rtol = 0.004 if fast_and_approximate_mode or device.arch() == ttnn.device.Arch.QUASAR else 0.002`

Comment (ign-sudha) [reply ttnn/cpp/ttnn/operations/reduction/generic/device/reduce_op_multi_core_h_program_factory.cpp:383]: Updated in the latest commit.

File: tests/ttnn/unit_tests/operations/reduce/test_reduction_mean.py [hunk 1]
```diff
@@ -176,3 +166,27 @@ def test_mean_shard(device, mem_config, keepdim, dtype):
...
+    torch_input_tensor = torch.rand(input_shape, dtype=torch.float32)
+    torch_output_tensor = torch.mean(torch_input_tensor.to(torch.float64)).to(torch.float32)
+
+    input_tensor = ttnn.from_torch(torch_input_tensor, layout=ttnn.TILE_LAYOUT, device=device, dtype=ttnn.float32)
+    input_tensor = ttnn.fill_implicit_tile_padding(input_tensor, TEST_PADDING_VALUE)
+
+    output_tensor = ttnn.mean(input_tensor, dim=None, fast_and_approximate_mode=fast_and_approximate_mode)
+    output_tensor = ttnn.to_torch(ttnn.from_device(output_tensor)).reshape(torch_output_tensor.shape)
+
+    if fast_and_approximate_mode or device.arch() == ttnn.device.Arch.QUASAR:
```
Comment (bbradelTT) [inline tests/ttnn/unit_tests/operations/reduce/test_reduction_mean.py:189]: Please create an issue to look at this once there are more SFPU LLKs supported on Quasar.

**Review by bbradelTT — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/reduce/test_reduction_mean.py (hunk 1 above)
Comment (ign-sudha) [reply tests/ttnn/unit_tests/operations/reduce/test_reduction_mean.py:189]: - #54145 is Created.
